### ISSUE TITLE
Link misses description

### ISSUE THREAD

**Issue by danielpeintner** [w3c/wot-thing-description#900]
A TD may contain multiple `links` pointing to arbitrary resources (array of `Link`).

The [Link](https://w3c.github.io/wot-thing-description/#link) itself misses though a way to describe what this link is for (e.g., a link to a "license" et cetera).

Hence I suggest to extend `Link` with an *additional* optional `description` field.

**Comment by sebastiankb** [comment]
to complaint with the web linking standard would `title` also work?

https://httpwg.org/specs/rfc8288.html#serialisation-defined-attributes

**Comment by egekorkan** [comment]
I don't think that this is the desired information by @danielpeintner . Title is mostly not for humans to read as a sentence but for machines to generate a button text. The standard above gives the example of menu item for example. However, for a UI generated based on a TD, both would be needed: title could generate the link/button text and description could generate the text displayed when mouse is over the button.

**Comment by vcharpenay** [comment]
in the specific example of "license", there is a [IANA registered relation type](https://www.iana.org/assignments/link-relations/link-relations.xhtml) you could use to express just that:

```json
{
  "href": "some license URI",
  "rel": "license"
}
```

One thing that can be done is to provide a list of relation types that are relevant for WoT from the IANA registry. These can then be added to the TD context (for instance, `license` could map to `http://purl.org/dc/terms/license`).

**Comment by vcharpenay** [comment]
If the description you'd like to have relates to the target resource itself (e.g. if you want to specify that the license is a MIT license) and not to the link itself, this description should be provided as a set of key/value pairs. That's what is called  _target attributes_ [in the Web linking](https://tools.ietf.org/html/rfc5988#section-5.4) spec.

I've looked at [CoRAL](https://tools.ietf.org/html/draft-ietf-core-coral-03), which is the "IETF companion" of the TD mode, w.r.t. hypermedia controls, and the CoRAL draft says: "a link in CoRAL does not have target attributes.  Instead, a link may have a list of zero or more nested elements.  These enable both the description of resource metadata and the chaining of links".

So, if we wanted to add target attributes to `Link` objects in a TD, we would need to add something to the model, I guess. Something like this (just indicative):

```json
{
  "href": "some license URI",
  "rel": "license",
  "targetAttributes": {
    "ex:licenseType": "MIT"
  }
}
```

**Comment by benfrancis** [comment]
@sebastiankb wrote:
> to complaint with the web linking standard would title also work?

+1.

But as @vcharpenay discusses, is the description meant to describe the link or the resource being linked to? What is the underlying use case you have in mind?

Perhaps you were just using it as en example, but regarding the link to a license, is the MIT license intended to apply to:
1. The content of the Thing Description
2. The software which generated the Thing Description
3. The physical device itself

I ask because none of the above make sense to me:
1. JSON resources fetched over the web are not usually licensed under a software license!
2. The software that generated the Thing Description is not being distributed and therefore the license is irrelevant to the client
3. Software licenses don't usually apply to physical devices either

**Comment by vcharpenay** [comment]
> I ask because none of the above make sense to me

it was maybe a poor choice, I didn't pay attention to what resource the license applied.
